flag.go

```diff
@@ -271,9 +271,18 @@ func stringifyFlag(f Flag) string {
 	needsPlaceholder := false
 	defaultValueString := ""
 	val := fv.FieldByName("Value")
+
+	showDefaultValue := true
+	if boolFlag, ok := f.(*BoolFlag); ok {
+		showDefaultValue = boolFlag.ShowDefaultValue
+	}
+
 	if val.IsValid() {
 		needsPlaceholder = val.Kind() != reflect.Bool
 		defaultValueString = fmt.Sprintf(formatDefault("%v"), val.Interface())
+		if !showDefaultValue {
+			defaultValueString = ""
+		}
```

Comment:
Can you invert this conditional please? Like so

```suggestion
		if showDefaultValue {
			defaultValueString = fmt.Sprintf(formatDefault("%v"), val.Interface())
		}
```